torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -237,12 +246,9 @@ class ProcessGroupNCCL : public ProcessGroup {
     // callback's stream and will have the result value of the callback.
     void addCallbackWithStream(
         std::function<void(void)> callback,
-        const c10::cuda::CUDAStream& stream,
-        std::shared_ptr<std::vector<at::cuda::CUDAEvent>> thenFutCudaEvents) {
+        const c10::cuda::CUDAStream& stream) {
       (*cudaEvents_)[0].block(stream);
-      c10::OptionalStreamGuard streamGuard{c10::Stream(stream)};
       callback();
-      (*thenFutCudaEvents)[0].record(stream);
     }
 
     // We use addCallbackWithStream instead of addCallback.
```

Comment:
We can just use `addCallback` here instead of `addCallbackWithStream` since we no longer need to pass the stream?